crates/sargon-uniffi/src/signing/rola_challenge.rs

```diff
@@ -0,0 +1,22 @@
+use crate::prelude::*;
+use sargon::RolaChallenge as InternalRolaChallenge;
+
+#[derive(Debug, Clone, PartialEq, InternalConversion, uniffi::Record)]
+pub struct RolaChallenge {
+    pub payload: BagOfBytes,
```

Comment:
Need
To include the whole ROLAChallege, including host and origin for Ledger HW

Reply:
Why? Those data are provided in `AuthenticationSigner`!

Reply:
Sure, depends on much back and forth one wants to do. But I think we want all sign input to be self contained.

On iOS we might be in global scope
When we impl the host
Interactor.
Thus we must send the full rolachallenge as sign input, since
ledger app requires it

Reply:
I'm trying to say that it might be two disjoint app contexts: where we from host call Sargon to ROLA Sign and where host is told by Sargon to sign ROLA

Reply:
I see what you mean, will do that

Reply:
Changed the `AuthenticationSigningInput` to contain the `nonce` and the `metadata`. We can get the `RolaChallenge` by doing:

```
input.rola_challenge()
```

With that way we can
1. Pass down to the host, the input that was requested
2. Deterministically derive the challenge using sargon os.
3. Validity checks still in place. For example checks the validity of the metadata's input url.